llms/mlx_lm/server.py

```diff
@@ -55,6 +55,26 @@ def stopping_criteria(
     return StopCondition(stop_met=False, trim_length=0)
 
 
+def sequence_overlap(s1: Sequence, s2: Sequence) -> int:
+    """
+    Check how much overlap two sequences have.
+        Only checks the end of s1 overlapping the start of s2
+
+    Args:
+        s1 (Sequence): The first sequence, which end is checked
+        s2 (Sequence): The second sequence, which beginning is checked
+
+    Returns:
+        int: The amount of overlap between s1 and s2
+    """
+    # Count down from the length of the smaller list -> Checks for larger overlaps first
```

Comment:
I'm wondering about the decision to start with larger overlaps? It seems like it could return faster if you start with smaller overlaps, but maybe I misunderstand how it's used?

Reply:
I had a version where I generated `len(stop_sequence) - overlap` tokens on a match, which required finding the longest match. But it overly complicated things for little benefit. So you have a good point, we can change it to check shorter matches first, and return a boolean instead.